src/frequenz/quantities/_marshmallow.py

```diff
@@ -0,0 +1,197 @@
+# License: All rights reserved
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Custom marshmallow fields and schema."""
+
+from typing import Any, Callable
+
+from marshmallow import EXCLUDE, Schema, fields
+
+from ._energy import Energy
+from ._percentage import Percentage
+from ._power import Power
+
+
+class PercentageField(fields.Field):
+    """A field for percentages."""
+
+    def _serialize(
+        self, value: Percentage, attr: str | None, obj: Any, **kwargs: Any
+    ) -> float:
+        """Serialize the percentage to a float.
+
+        Args:
+            value: the percentage
+            attr: the attribute name
+            obj: the object
+            **kwargs: additional keyword arguments
+
+        Returns:
+            The percentage as a float
+        """
+        return value.as_percent()
+
+    def _deserialize(
+        self, value: float, attr: str | None, data: Any, **kwargs: Any
+    ) -> Percentage:
+        """Deserialize the percentage from a float.
+
+        Args:
+            value: the float
+            attr: the attribute name
+            data: the data
+            **kwargs: additional keyword arguments
+
+        Returns:
+            The percentage
+        """
+        return Percentage.from_percent(value)
+
+
+class EnergyField(fields.Field):
+    """A field for energies."""
+
+    def _serialize(
+        self, value: Energy, attr: str | None, obj: Any, **kwargs: Any
+    ) -> float:
+        """Serialize the energy to a float.
+
+        Args:
+            value: the energy
+            attr: the attribute name
+            obj: the object
+            **kwargs: additional keyword arguments
+
+        Returns:
+            The energy as a float
+        """
+        return value.as_watt_hours()
+
+    def _deserialize(
+        self, value: float, attr: str | None, data: Any, **kwargs: Any
+    ) -> Energy:
+        """Deserialize the energy from a float.
+
+        Args:
+            value: the float
+            attr: the attribute name
+            data: the data
+            **kwargs: additional keyword arguments
+
+        Returns:
+            The energy
+        """
+        return Energy.from_watt_hours(value)
```

Comment:
I wouldn't hard code the used unit. I would actually parse from a str, so the user can set this to `100kWh` or `-10mWh`. As for serialization, just also `str()` should do the trick.

Reply:
We could change that, but wouldn't wanna do that in this PR.
Besides it would require that we use strings in the config and I wasn't willing to have
`power = "10 kw"` instead of `power_kw = 10` if I can avoid it

Reply:
OK, I see that point, but I'm not sure. We can't enforce people to add the unit to variable names but we can enforce it in the value, so it should be less error prone, at the expense of  not being able to detect other formatting more easily with a linter (for example `power = "1;0 kw"` won't be shown as an error but `power = 1;0` probably will).

Another advantage of having the unit in the value is for cases where the range is not so clear, like timeouts, where you can end up with stuff like `time_s = 0.01` or `time_ms  = 5000` very easily, which is harder to read than `time = "10ms"` or `time = "5s"`.

But yeah, there are many trade-offs and maybe there is no clear winner, so if we add it, we could add it as an option (parse this field as a string, or as watt or as megawatt), but I agree this is for a different PR.

Reply:
>we could add it as an option (parse this field as a string, or as watt or as megawatt), but I agree this is for a different PR.

well, too late,